src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -191,18 +191,20 @@ class FeelInterpreter(private val valueMapper: ValueMapper) {
             p => allValues(p.map(vars => () => eval(condition)(context.addAll(vars))), ValBoolean)
           )
         )
-      case For(iterators, exp)             =>
+      case For(iterators, exp) =>
         withValOrNull(
           withCartesianProduct(
             iterators,
-            p =>
-              ValList((List[Val]() /: p) {
-                case (partial, vars) => {
-                  val iterationContext = context.addAll(vars).add("partial" -> ValList(partial))
-                  val value            = eval(exp)(iterationContext)
-                  partial ++ (value :: Nil)
-                }
-              })
+            p => {
+              var partial = Vector.empty[Val]
```

Comment:
:thought_balloon: usually for better performances it's advised to use `Vector.newBuilder[Val]` which creates a mutable builder optimized for constructing the `Vector`.  You must call `builder.build` at the end to get the final vector. 
This with the caveat that it's ok to use a mutable builder instead of an immutable one

Reply:
I think in this case, it doesn't help because we immediately need the built Vector again to pass it in the context.